### DISCUSSION TITLE
Improved iOS demo app

### DISCUSSION THREAD

**Discussion by axxel** [zxing-cpp/zxing-cpp#639] [Ideas]
First I have an obviously trivial question for those who know Swift: How does [this line](https://github.com/zxing-cpp/zxing-cpp/blob/893b8c25dd7a8cb1e71a30a67db94cec7e9891e4/wrappers/ios/demo/demo/ViewController.swift#L73) map to [this interface](https://github.com/zxing-cpp/zxing-cpp/blob/master/wrappers/ios/Sources/Wrapper/Reader/ZXIBarcodeReader.h#L12)? There is no function called `read`. Is this some implicit function overloading magic happening between ObjectiveC and Swift? If so, would this also be relevant for the recent `writeBytes`/`writeText` addition?

Would someone be interested in pimping the demo code a little bit, like:
 * make use of the `DecodeHints` feature to showcase it
 * paint some lines over the image according to the `position` of the result
 * maybe add support for actually showing all, not only the first detected symbol

Any other ideas how to make the demo more helpful to get people started?

@benjohnde, @alexmanzer

**Comment by alexmanzer** [comment]
> First I have an obviously trivial question for those who know Swift: How does [this line](https://github.com/zxing-cpp/zxing-cpp/blob/893b8c25dd7a8cb1e71a30a67db94cec7e9891e4/wrappers/ios/demo/demo/ViewController.swift#L73) map to [this interface](https://github.com/zxing-cpp/zxing-cpp/blob/master/wrappers/ios/Sources/Wrapper/Reader/ZXIBarcodeReader.h#L12)? There is no function called read. Is this some implicit function overloading magic happening between ObjectiveC and Swift? If so, would this also be relevant for the recent writeBytes/writeText addition?

@axxel Very well noticed. Embarrassing that I didn't think of that. Soon you'll be the Swift pro here. 😃
Since this is IMO the only right thing (especially when looking at the `read` interface), I have created a PR for this change:
https://github.com/zxing-cpp/zxing-cpp/pull/640
